src/rgb/schema/field.rs

```diff
@@ -32,6 +34,35 @@ pub enum FieldFormat {
     Signature(SignatureAlgorithm),
 }
 
+impl FieldFormat {
+    pub fn validate(&self, value: &Value) -> Result<(), ValidationError> {
+        match (self, value) {
```

Comment:
i personally prefer to keep them this way to avoid mistaking enum variants for structs, but i can adapt if the rest of the code follows different rules. keep in mind that this is still a very very early draft, so many things will change before it's ready to be merged